micro-rdk/src/common/ota.rs

```diff
@@ -352,53 +363,97 @@ impl<S: OtaMetadataStorage> OtaService<S> {
+                            retry = true;
                         }
                     }
                 }
                 Err(e) => {
                     log::error!("failed to create http connection: {}", e);
+                    retry = true;
                 }
             };
 
-            log::debug!(
-                "retry establishing http connection to {} in {} seconds",
-                &uri,
-                CONN_RETRY_SECS
-            );
-            Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
-        };
-
-        let conn = Box::new(self.exec.spawn(async move {
-            if let Err(err) = conn.await {
-                log::error!("connection failed: {:?}", err);
+            if sender.is_none() || inner_conn.is_none() {
+                return Err(OtaError::Other(
+                    "invalid state, error path not handled".to_string(),
+                ));
+            }
+            let mut sender = sender.unwrap();
+            let inner_conn = inner_conn.unwrap();
+
+            if retry {
+                log::debug!(
+                    "retry establishing http connection to {} in {} seconds",
+                    &uri,
+                    CONN_RETRY_SECS
+                );
+                Timer::after(Duration::from_secs(CONN_RETRY_SECS)).await;
+                continue;
             }
-        }));
-
-        log::info!("ota connected, beginning download");
-        let request = Request::builder()
-            .method("GET")
-            .uri(uri)
-            .body(Empty::<Bytes>::new())
-            .map_err(|e| OtaError::Other(e.to_string()))?;
-        let response = sender
-            .send_request(request)
-            .await
-            .map_err(|e| OtaError::Other(e.to_string()))?;
 
-        // TODO(RSDK-9346): handle other success codes
-        if response.status() != 200 {
-            return Err(OtaError::Other(
-                format!("Bad Request - Status:{}", response.status()).to_string(),
-            ));
+            conn = Some(Box::new(self.exec.spawn(async move {
+                if let Err(err) = inner_conn.await {
+                    log::error!("connection failed: {:?}", err);
+                }
+            })));
+
+            log::info!("ota connected, beginning download");
+            let request = Request::builder()
+                .method("GET")
+                .uri(uri)
+                .body(Empty::<Bytes>::new())
+                .map_err(|e| OtaError::Other(e.to_string()))?;
+            response = sender
+                .send_request(request)
+                .await
+                .map_err(|e| OtaError::Other(e.to_string()))?;
+
+            let status = response.status();
+            match (status.is_success(), status.is_redirection()) {
+                (true, false) => break,
+                (false, true) => {
+                    log::debug!("target url has been redirected...");
```

Comment:
[optional] I'd make this `info` level, Say `OTA` in it, and give the original URL